Open Judge System/Web/OJS.Web/Controllers/SubmissionsController.cs

```diff
@@ -66,7 +67,20 @@ public ActionResult Index()
             bool notProcessedOnly = false,
             int? contestId = null)
         {
-            var data = this.User.IsAdmin() ? this.Data.Submissions.All() : this.Data.Submissions.AllPublic();
+            IQueryable<Submission> data = null;
+
+            if (this.User.IsLecturer())
+            {
+                data = this.Data.Submissions.AllForLecturer(this.UserProfile.Id);
```

Comment:
Can we change the name of the method to something more descriptive like AllPublicWithLecturerContests?